docs/azure_arc_jumpstart/azure_arc_k8s/day2/cluster_api/cluster_api_keyvault_extension/_index.md

```diff
@@ -0,0 +1,142 @@
+
+For you to get familiar with the automation and deployment flow, below is an explanation.
+
+* User has deployed Kubernetes using Cluster API and has it connected as Azure Arc-enabled Kubernetes cluster.
+
+* User is editing the environment variables on the Shell script file (1-time edit) which then be used throughout the extension deployment.
+
+* User is running the shell script. The script will use the extension management feature of Azure Arc to deploy the Azure Key Vault Secrets Provider extension on the Azure Arc connected cluster.
+
+* The script will also deploy the sample app to Azure Arc-enabled Kubernetes cluster
+
+* User is veryfing the cluster and make sure the extension is deployed.
+
+* User is checking the secret value mounted in pod.
+
+## Create Azure Key Vault extensions instance
+
+To create a new extension Instance, we will use the _k8s-extension create_ command while passing in values for the mandatory parameters. This scenario provides you with the automation to deploy the Azure Key Vault extension on your Azure Arc-enabled Kubernetes cluster.
+
+> **Note: Before installing the Key Vault extension, make sure that the kubectl context is pointing to your Azure Arc-enabled Kubernetes cluster. To do that, you can refer to the [official Kubernetes documentation](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) to find the options to change the kubecontext to different Kubernetes clusters.**
+
+![Screenshot showing current kubectl context pointing to CAPI cluster](./01.png)
+
+* In the screenshot below, notice how currently there are no extensions installed yet in your Arc-enabled Kubernetes cluster.
+
+    ![Screenshot showing Azure Portal with Azure Arc-enabled Kubernetes resource extensions](./02.png)
+
+* Edit the environment variables in the script to match your environment parameters followed by running the ```. ./azure_keyvault_k8s_extension.sh``` command.
+
+    ![Screenshot parameter examples](./03.png)
+
+    > **Note: The extra dot is due to the shell script has an *export* function and needs to have the vars exported in the same shell session as the rest of the commands.**
+
+   The script will:
+
+  * Login to your Azure subscription using the SPN credentials
+  * Add or Update your local _connectedk8s_ and _k8s-extension_ Azure CLI extensions
+  * Create Resource Group, Key Vault and sample secret
+  * Create Azure Key Vault k8s extension instance
+  * Create Kubernetes Secret for supporting SPN Authentication
+  * Create Secret Provider Class to read the secrets
+  * Deploy the app
+
+* You can now see that Azure Key Vault extensions are now enabled in the extension tab section of the Azure Arc-enabled Kubernetes cluster resource in Azure.
+
+    ![Screenshot extension deployment extensions tab](./04.png)
+
+* You can also verify the deployment by running the below _kubectl_ commands and see the deployed artifacts in the _kube-system_ namespace.
```

Comment:
Please add the commands to run as text values as well using

```shell
```

Reply:
Added the commands